### PR TITLE
Replace CoreOS with Flatcar

### PR DESCRIPTION
**Pull request by corest**

### CODE REVIEW

**Review by calvix — APPROVED** [review]

File: config.yaml.dist [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # used to install machines and if no coreos
 # version is given in the profile
-default_coreos_version: 1409.7.0
+default_flatcar_version: 1409.7.0
```
Comment (kopiczko) [inline config.yaml.dist:3]: Not 2543.3.0?

File: scripts/fetch-flatcar-image [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+IMAGES_DIR=${IMAGES_DIR:-$DEFAULT_IMAGES_DIR}
+KEEP_IMAGES=${KEEP_IMAGES:-"false"}
+
+if [ ${KEEP_IMAGES} = "false" ]; then
+    # remove old images
+    rm -rf ${IMAGES_DIR}/${FLATCAR_VERSION}
+fi
+
+mkdir -p ${IMAGES_DIR}/${FLATCAR_VERSION}
+
```
Comment (kopiczko) [inline scripts/fetch-flatcar-image:18]: Why not:

```sh
if [ -f ${IMAGES_DIR}/${FLATCAR_VERSION}/keep ]; then
    echo "Images for ${FLATCAR_VERSION} already exist on filesystem"
    exit 0
fi
```

File: scripts/fetch-flatcar-image [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    echo "$FLATCAR_VERSION" > ${IMAGES_DIR}/${FLATCAR_VERSION}/flatcar-version
+
+    mkdir -p ${IMAGES_DIR}/${FLATCAR_VERSION}/etc/systemd/system/ignition-disks.service.d/
+    cp inject_pxe/00-ignition-delay.conf ${IMAGES_DIR}/${FLATCAR_VERSION}/etc/systemd/system/ignition-disks.service.d/
+
+    docker run  --net=host --rm -v ${IMAGES_DIR}/${FLATCAR_VERSION}:/usr/code/images \
+        alpine:3.8 /bin/sh -c "apk add cpio gzip && \
+        zcat /usr/code/images/flatcar_pxe_image.cpio.gz > /usr/code/images/flatcar_production_pxe_image.cpio && \
+        cd /usr/code/images && find etc | cpio -o -A -H newc -O flatcar_production_pxe_image.cpio && \
+        gzip -f flatcar_production_pxe_image.cpio && echo DONE"
```
Comment (kopiczko) [inline scripts/fetch-flatcar-image:32]: For my interest: what does this do?

Comment (glitchcrab) [reply scripts/fetch-flatcar-image:32]: If I remember rightly, it's just unpacking the Flatcar image, dropping the `00-ignition-delay.conf` file into the filesystem and then building/compressing the image again.

Comment (kopiczko) [reply scripts/fetch-flatcar-image:32]: If even you don't know then maybe dropping a comment would be an idea.

File: config.yaml.dist (hunk 1 above)
Comment (corest) [reply config.yaml.dist:3]: It is just example

**Comment by corest** [comment]
This PR is not about refactoring, just updating references. If we want to improve it - we can have an issue for it

**Comment by corest** [comment]
Tested in hive ci. Works fine. Merging

File: scripts/fetch-flatcar-image (hunk 3 above)
Comment (calvix) [reply scripts/fetch-flatcar-image:32]: decompress image, add our custom file and compress it back

**Pull request merged**